src/components/main/files/mod.rs

```diff
@@ -15,6 +14,8 @@ use utils::DRAG_FILE_EVENT;
 #[cfg(target_os = "windows")]
 use dioxus::desktop::wry::webview::FileDropEvent;
 
+pub static FILES_STATE: AtomRef<Vec<Directory>> = |_| Vec::new();
```

Comment:
Any reason to why not use `Constellation::get_path` to grab the path (or depth) it is in the index over the vector `Directory`? From what I understand from our previous discussion, youre using this to track the path so when changing over it would be in the last directory it was in. Constellation does keep track of the internal path so that doesnt exactly reset and could grab it, split it to represent the directories names.